src/atomate2/lobster/schemas.py

```diff
@@ -0,0 +1,585 @@
+        )  # initialize lobsterpy condensed bonding analysis
+        cba_cohp_plot_data = {}  # Initialize dict to store plot data
+
+        set_cohps = analyse.set_cohps
+        set_labels_cohps = analyse.set_labels_cohps
+        set_inequivalent_cations = analyse.set_inequivalent_ions
+        struct = analyse.structure
+        for _iplot, (ication, labels, cohps) in enumerate(
+            zip(set_inequivalent_cations, set_labels_cohps, set_cohps)
+        ):
+            namecation = str(struct[ication].specie)
+            for label, cohp in zip(labels, cohps):
+                if label is not None:
+                    cba_cohp_plot_data.update(
+                        {
+                            namecation
+                            + str(ication + 1)
+                            + ": "
+                            + label: {
+                                "COHP": list(cohp.get_cohp()[Spin.up]),
+                                "ICOHP": list(cohp.get_icohp()[Spin.up]),
+                                "Energies": list(cohp.energies),
+                                "Efermi": cohp.efermi,
+                            }
+                        }
+                    )
+
+        describe = Description(analysis_object=analyse)
+
+        lpca = CondensedBondingAnalysis(
+            formula=cba["formula"],
+            max_considered_bond_length=cba["max_considered_bond_length"],
+            limit_icohp=cba["limit_icohp"],
+            number_of_considered_ions=cba["number_of_considered_ions"],
+            sites=cba["sites"],
+            type_charges=analyse.type_charge,
+            cohp_plot_data=cba_cohp_plot_data,
+            cutoff_icohp=analyse.cutoff_icohp,
+            summed_spins=True,
+            which_bonds=analyse.whichbonds,
+            final_dict_bonds=analyse.final_dict_bonds,
+            final_dict_ions=analyse.final_dict_ions,
+            run_time=cba_run_time,
+        )
+
+        lobster_out = LobsteroutModel(**lobsterout_doc)
+        lobster_in = LobsterinModel(**lobsterin_here)
+
+        charge = Charge(os.path.join(dir_name, "CHARGE.lobster.gz"))
+
+        charges = {"Mulliken": charge.Mulliken, "Loewdin": charge.Loewdin}
+
+        icohplist = Icohplist(
+            filename=os.path.join(dir_name, "ICOHPLIST.lobster.gz"),
+            are_cobis=False,
+            are_coops=False,
+        )
+        icobilist = Icohplist(
+            filename=os.path.join(dir_name, "ICOBILIST.lobster.gz"),
+            are_cobis=True,
+            are_coops=False,
+        )
+        icooplist = Icohplist(
+            filename=os.path.join(dir_name, "ICOOPLIST.lobster.gz"),
+            are_coops=True,
+            are_cobis=False,
+        )
+
+        icohp_dict = icohplist.icohpcollection.as_dict()
+        icobi_dict = icobilist.icohpcollection.as_dict()
+        icoop_dict = icooplist.icohpcollection.as_dict()
+
+        def get_strng_bonds(bondlist, are_cobis, are_coops, relevant_bonds: dict):
```

Comment:
The `from_directory` is already quite long. Can you put this as a private method outside of `from_directory` to make this a bit easier to read?

Reply:
Refactored.